### ISSUE TITLE
2016.11 breaks file.managed on Windows

### ISSUE THREAD

**Issue by lorengordon** [saltstack/salt#38443]
### Description of Issue/Question

Some changes were made in #37368 to `file.extrach_hash` which introduce a dependency on the `string` library. The necessary import was not added to modules/win_file.py, so any call to that function throws an exception:

```
[DEBUG   ] Traceback (most recent call last):
  File "C:\salt\bin\lib\site-packages\salt\states\file.py", line 1835, in managed
    **kwargs
  File "C:\salt\bin\lib\site-packages\salt\modules\file.py", line 3720, in get_managed
    saltenv)
  File "C:\salt\bin\lib\site-packages\salt\modules\file.py", line 636, in get_source_sum
    ret = extract_hash(hash_fn, '', file_name, source, source_hash_name)
  File "C:\salt\bin\lib\site-packages\salt\modules\file.py", line 3858, in extract_hash
    filename_separators = string.whitespace + r'\/'
NameError: global name 'string' is not defined
```

### Versions Report
```
PS C:\> C:\salt\salt-call.bat --versions
Salt Version:
           Salt: 2016.11.1

Dependency Versions:
           cffi: 1.7.0
       cherrypy: 7.1.0
       dateutil: 2.5.3
          gitdb: 0.6.4
      gitpython: 2.0.8
          ioflo: 1.5.5
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: 1.4.5
       M2Crypto: Not Installed
           Mako: 1.0.4
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (v2.7.12:d33e0cf91556, Jun 27 2016, 15:24:40) [MSC v.1500 64 bit (AMD64)]
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 16.0.1
           RAET: Not Installed
          smmap: 0.9.0
        timelib: 0.2.4
        Tornado: 4.4.1
            ZMQ: 4.1.6

System Versions:
           dist:
        machine: AMD64
        release: 2012ServerR2
         system: Windows
        version: 2012ServerR2 6.3.9600  Multiprocessor Free
```

**Comment by mbochenk** [comment]
Exactly the same behaviour on both 2016.11.0 and 2016.11.1. This seems a rather critical one.
```
          ID: copy-glassfish-zip
    Function: file.managed
        Name: C:\Users\ADMINI~1\AppData\Local\Temp\2\glassfish-4.1.1.zip
      Result: False
     Comment: Unable to manage file: global name 'string' is not defined
     Started: 01:33:29.390000
    Duration: 31.0 ms
     Changes:
```

```
c:\salt>salt-call --versions-report
Salt Version:
           Salt: 2016.11.0

Dependency Versions:
           cffi: 1.7.0
       cherrypy: 7.1.0
       dateutil: 2.5.3
          gitdb: 0.6.4
      gitpython: 2.0.8
          ioflo: 1.5.5
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: 1.4.5
       M2Crypto: Not Installed
           Mako: 1.0.4
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (v2.7.12:d33e0cf91556, Jun 27 2016, 15:24:40) [MSC v.150
0 64 bit (AMD64)]
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 16.0.1
           RAET: Not Installed
          smmap: 0.9.0
        timelib: 0.2.4
        Tornado: 4.4.1
            ZMQ: 4.1.6

System Versions:
           dist:
        machine: AMD64
        release: 2008ServerR2
         system: Windows
        version: 2008ServerR2 6.1.7601 SP1 Multiprocessor Free

```

**Comment by lorengordon** [comment]
@mbochenk fix was just merged into the 2016.11 branch. i'd expect it to make it into the 2016.11.2 release. issue will close automatically when the commit is merged forward into the develop branch.
